traitlets/traitlets.py

```diff
@@ -486,15 +486,14 @@ def _dynamic_default_callable(self, obj):
         return getattr(self, 'make_dynamic_default', None)
 
     def instance_init(self, obj):
-        obj._cross_validation_lock = True
```

Comment:
This is the block that made me think this change was worth making.  If something inside the block fails, obj._cross_validation_lock will be stuck on `True`.  

I think this happens to be harmless here, because we're assuming that we're being called from `obj.__new__`, which means an uncaught exception will prevent a live instance to the invalid object from ever existing, but repeating this pattern elsewhere would result in hard-to-find bugs where the relative ordering of validators would cause very different behaviors.